lib/dash/dash_parser.js

```diff
@@ -410,6 +410,12 @@ shaka.dash.DashParser.prototype.requestManifest_ = function() {
       return;
     }
 
+    // For redirections add the response uri to the first entry in the
+    // Manifest Uris array.
+    if (response.uri && this.manifestUris_.indexOf(response.uri) == -1) {
```

Comment:
One small nit: you can use ES6's includes() method on Arrays now.  For example, && !this.manifestUris_.includes(response.uri)